k8s/upgrade/src/bin/upgrade-job/common/kube_client.rs

```diff
@@ -48,9 +59,28 @@ pub(crate) struct KubeClientSet {
     pods_api: Api<Pod>,
     namespaces_api: Api<Namespace>,
     deployments_api: Api<Deployment>,
+    crds_api: Api<CustomResourceDefinition>,
 }
 
 impl KubeClientSet {
+    pub(crate) async fn get() -> Self {
+        KUBE_CLIENT_SET_BUILDER
```

Comment:
Why do this?

Reply:
Because, the kube-client set has clients for namespaced kubernetes resources.

Previously we have been creating a client whenever one is required. Because of the namespaced resources, functions which need a client but not necessarily the namespaced resources, have to accept the namespace as an argument.

With a global client builder, the client set builder is always available, and it always knows about the namespace. And this frees up these functions from having to know about the namespace.

Reply:
From what I can tell the namespace is available on most places where it's used?
Most times I've used these singleton patterns I later came to regret it. I suggest you rename get to more thing that makes it clearer to the caller that the namespace is pre-configured from something else. Or have the namespace itself as a singleton.

Reply:
Makes sense to me. I've tossed the whole thing out for a simpler client module.